semgrep/semgrep/semgrep_main.py

```diff
@@ -260,13 +261,16 @@ def main(
 
     output_handler.handle_semgrep_errors(semgrep_errors)
 
-    rule_matches_by_rule = {
-        rule: [
-            attr.evolve(rule_match, is_ignored=rule_match_nosem(rule_match, strict))
-            for rule_match in rule_matches
-        ]
-        for rule, rule_matches in rule_matches_by_rule.items()
-    }
+    nosem_errors = []
+    for rule, rule_matches in rule_matches_by_rule.items():
+        evolved_rule_matches = []
+        for rule_match in rule_matches:
+            ignored, returned_errors = rule_match_nosem(rule_match, strict)
+            evolved_rule_matches.append(attr.evolve(rule_match, is_ignored=ignored))
+            nosem_errors.extend(returned_errors)
+        rule_matches_by_rule[rule] = evolved_rule_matches
+
+    output_handler.handle_semgrep_errors(nosem_errors)
```

Comment:
Should we instead append to `semgrep_errors` and make one call to `handle_semgrep_errors`? Not sure if that matters in our `output_handler` error handling.

Reply:
Doesn't matter functionality-wise. Seems ok to add it to its own list and call handle_semgrep_errors twice